src/System.Runtime/tests/System/Reflection/AssemblyTests.cs

```diff
@@ -21,8 +23,29 @@
 
 namespace System.Reflection.Tests
 {
-    public static class AssemblyTests
+    public class AssemblyTests : IDisposable
     {
+
+        string sourceTestAssemblyPath = Path.Combine(Environment.CurrentDirectory, "TestAssembly.dll");
+        string destTestAssemblyPath = Path.Combine(Environment.CurrentDirectory, "TestAssembly", "TestAssembly.dll");
+
+        public AssemblyTests()
+        {
+            // Move TestAssembly.dll to subfolder TestAssembly
+            if(!File.Exists(destTestAssemblyPath))
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(destTestAssemblyPath));
+                File.Move(sourceTestAssemblyPath, destTestAssemblyPath);
+            }
+        }
+
+        public void Dispose()
+        {
+            // Revert TestAssembly.dll back to its previous location
+            if(!File.Exists(sourceTestAssemblyPath))
```

Comment:
although the file is loaded move seems to be working fine on my machine and also in CI.